samples/bluetooth/peripheral/src/main.c

```diff
@@ -265,6 +262,33 @@ static void bt_ready(void)
 		settings_load();
 	}
 
+	err = bt_gatt_get_svc_uuid_data(BT_DATA_UUID16_ALL, &ad[1],
+					uuid16_buf, sizeof(uuid16_buf));
+	if (err != 0) {
+		printk("Could not encode all UUID16 data: %d\n", err);
+
+		err = bt_gatt_get_svc_uuid_data(BT_DATA_UUID16_SOME, &ad[1],
+						uuid16_buf, sizeof(uuid16_buf));
+		if (err != 0) {
+			printk("Could not encode some UUID16 data: %d\n", err);
+			return;
+		}
+	}
+
+	err = bt_gatt_get_svc_uuid_data(BT_DATA_UUID128_ALL, &ad[2],
+					uuid128_buf, sizeof(uuid128_buf));
+	if (err != 0) {
+		printk("Could not encode UUID128 data: %d\n", err);
+
+		err = bt_gatt_get_svc_uuid_data(BT_DATA_UUID128_SOME, &ad[2],
+						uuid128_buf,
+						sizeof(uuid128_buf));
+		if (err != 0) {
+			printk("Could not encode some UUID16 data: %d\n", err);
+			return;
+		}
+	}
```

Comment:
An alternative I considered was to automatically change the type to _SOME instead of _ALL if it did not fit, instead of returning an error and letting the application decided what to do.